xbmc/guilib/GUIFontCache.cpp

```diff
@@ -75,27 +95,26 @@ class CGUIFontCacheImpl
 };
 
 template<class Position, class Value>
-void CGUIFontCacheEntry<Position, Value>::Reassign::operator()(CGUIFontCacheEntry<Position, Value> &entry)
+CGUIFontCacheEntry<Position, Value>::~CGUIFontCacheEntry()
 {
-  entry.m_key.m_pos = m_key.m_pos;
-  entry.m_key.m_colors.assign(m_key.m_colors.begin(), m_key.m_colors.end());
-  entry.m_key.m_text.assign(m_key.m_text.begin(), m_key.m_text.end());
-  entry.m_key.m_alignment = m_key.m_alignment;
-  entry.m_key.m_maxPixelWidth = m_key.m_maxPixelWidth;
-  entry.m_key.m_scrolling = m_key.m_scrolling;
-  entry.m_matrix = m_key.m_matrix;
-  entry.m_key.m_scaleX = m_key.m_scaleX;
-  entry.m_key.m_scaleY = m_key.m_scaleY;
-
-  entry.m_lastUsedMillis = m_nowMillis;
-  entry.m_value.clear();
+  delete &m_key.m_colors;
+  delete &m_key.m_text;
+  m_value.clear();
 }
 
 template<class Position, class Value>
-CGUIFontCacheEntry<Position, Value>::~CGUIFontCacheEntry()
+void CGUIFontCacheEntry<Position, Value>::Assign(const CGUIFontCacheKey<Position> &key, unsigned int nowMillis)
 {
-  delete &m_key.m_colors;
-  delete &m_key.m_text;
```

Comment:
should these really be removed? Assign is called on an existing entry so this could be a memory leak